services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/plugins/S3DisableChunkEncodingIfConfiguredPlugin.java

```diff
@@ -0,0 +1,93 @@
+ * {@link S3AuthSchemeProvider} that sets {@link  AwsV4FamilyHttpSigner#CHUNK_ENCODING_ENABLED} to false.
+ */
+@SdkInternalApi
+public final class S3DisableChunkEncodingIfConfiguredPlugin implements SdkPlugin {
+
+    private static final Logger LOG = Logger.loggerFor(S3DisableChunkEncodingIfConfiguredPlugin.class);
+
+    private final boolean isServiceConfigurationPresent;
+    private final boolean isChunkedEncodingEnabledConfigured;
+    private final boolean isChunkedEncodingEnabledDisabled;
+    private final boolean configuresDisableChunkEncoding;
+
+    public S3DisableChunkEncodingIfConfiguredPlugin(SdkClientConfiguration config) {
+        S3Configuration serviceConfiguration =
+            (S3Configuration) config.option(SdkClientOption.SERVICE_CONFIGURATION);
+
+        boolean isServiceConfigurationPresent = serviceConfiguration != null;
+        boolean shouldAddDisableChunkEncoding = false;
+        boolean isChunkedEncodingEnabledConfigured = false;
+        boolean isChunkedEncodingEnabledDisabled = false;
+        boolean configuresDisableChunkEncoding = false;
+        if (isServiceConfigurationPresent) {
+            isChunkedEncodingEnabledConfigured = serviceConfiguration.toBuilder().chunkedEncodingEnabled() != null;
+            isChunkedEncodingEnabledDisabled = !serviceConfiguration.chunkedEncodingEnabled();
+            configuresDisableChunkEncoding = isChunkedEncodingEnabledConfigured && isChunkedEncodingEnabledDisabled;
+            if (configuresDisableChunkEncoding) {
+                shouldAddDisableChunkEncoding = true;
+            }
+        }
+        this.configuresDisableChunkEncoding = shouldAddDisableChunkEncoding;
+        this.isChunkedEncodingEnabledConfigured = isChunkedEncodingEnabledConfigured;
+        this.isChunkedEncodingEnabledDisabled = isChunkedEncodingEnabledDisabled;
+        this.isServiceConfigurationPresent = isServiceConfigurationPresent;
+    }
+
+    @Override
+    public void configureClient(SdkServiceClientConfiguration.Builder config) {
+        if (configuresDisableChunkEncoding) {
+            LOG.debug(() -> String.format("chunkedEncodingEnabled was explicitly disabled in the configuration, adding "
+                                          + "`S3DisableChunkEncodingAuthSchemeProvider` auth provider wrapper."));
+            S3ServiceClientConfiguration.Builder s3Config = (S3ServiceClientConfiguration.Builder) config;
+
+            // We wrap the default S3AuthSchemeProvider with a S3DisableChunkEncodingAuthSchemeProvider
+            // instance that sets the AwsV4FamilyHttpSigner.CHUNK_ENCODING_ENABLED signer
+            // property to false but only if the operations is `"PutObject" or "UploadPart"`
+            // This legacy logic was implemented before using an interceptor but now requires
+            // wrapping the S3AuthSchemeProvider for it to work.
```

Comment:
Can we remove this comment and maybe add the legacy part to the class doc instead?